client/components/banner/index.jsx

```diff
@@ -91,34 +104,40 @@ export class Banner extends Component {
 	}
 
 	handleClick = e => {
-		const { event, feature, onClick } = this.props;
+		const { event, feature, compact, onClick, tracksClickName, tracksClickProperties } = this.props;
 
-		if ( event ) {
-			this.props.recordTracksEvent( 'calypso_banner_cta_click', {
-				cta_name: event,
-				cta_feature: feature,
-				cta_size: 'regular',
-			} );
+		if ( ! tracksClickName ) {
+			return;
 		}
 
+		this.props.recordTracksEvent( tracksClickName, {
+			cta_name: event,
+			cta_feature: feature,
+			cta_size: compact ? 'compact' : 'regular',
+			...tracksClickProperties,
+		} );
+
 		onClick( e );
 	};
 
 	handleDismiss = e => {
-		const { event, feature, onDismiss } = this.props;
+		const { event, feature, onDismiss, tracksDismissName, tracksDismissProperties } = this.props;
 
-		if ( event ) {
-			this.props.recordTracksEvent( 'calypso_banner_dismiss', {
-				cta_name: event,
-				cta_feature: feature,
-			} );
+		if ( ! tracksDismissName ) {
```

Comment:
Same as above here, we should keep executing `onDismiss`  even if tracking is opted-out. And let's also keep the check that verifies if `event` is defined to ensure backwards compatibility:

```js
if ( tracksDismissName && event ) {
	this.props.recordTracksEvent( tracksDismissName, {
		cta_name: event,
		cta_feature: feature,
		...tracksDismissProperties,
	} );
}

onDismiss( e );
```